src/server/gitHubSearch.ts

```diff
@@ -0,0 +1,25 @@
+import fetch from "node-fetch";
+
+type ErrorResult = {
+  message: string;
+};
+
+export async function fetchGitHubSearchCodeWithUser<T>(
```

Comment:
This generic `<T>` is unused and doesn't make sense here.
Instead, try to create the return type based on the structure you get from GitHub (or don't add a return type for now).